compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/AbstractLexer.java

```diff
@@ -306,4 +306,8 @@ protected void processNumericEscape() {
 
         this.reader.advance();
     }
+
+    protected static boolean isWhitespace(int ch) {
+        return (ch == 0x0020 || ch == 0x0009 || ch == 0x000A || ch == 0x000D);
+    }
```

Comment:
some set of cases are missing here I think we need to do what `processSyntaxTrivia` does only using peek

Reply:
```bal
function main() {
    string:RegExp b = re // some documentation -----------------  string
    ``;
}
```
this should be allowed

Reply:
I will update that

Reply:
Wondering whether we have a function to reuse already?

Reply:
There is no exiting function to handle this scenario. Created this function https://github.com/ballerina-platform/ballerina-lang/pull/38495/files#diff-492a9dcba142de0e98b2627fc2c2d530491cb8471aca5c74be0b8060d0c69639R1052